File: Backend/ClothingStore.Core/Models/Product/ProductFilterDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClothingStore.Core.Models.Product
{
    public class ProductFilterDTO
    {
        public string? Keyword { get; set; }          
        public string? Brand { get; set; }            
        public decimal? MinPrice { get; set; }        
        public decimal? MaxPrice { get; set; }        
        public bool? InStockOnly { get; set; }        
        public string? Color { get; set; }            
        public string? Size { get; set; }             
        public string? SortBy { get; set; }           
    }

}

File: Frontend/src/models/product/product-filter-dto.ts
export interface ProductFilterDto {
  keyword?: string;
  minPrice?: number;
  maxPrice?: number;
  brand?: string;
  size?: string;
  color?: string;
  inStockOnly?: boolean;
  sortBy?: 'PriceAsc' | 'PriceDesc' | 'Newest' | 'Oldest';
}